.github/workflows/tests.yml

```diff
@@ -1,16 +1,19 @@
 name: Tests
 
-on: [push]
+on: [push, pull_request]
+
+env:
+  PYTHON_VERSIONS: ["3.8", "3.9", "3.10", "3.11", "3.12", "pypy-3.8", "pypy-3.9", "pypy-3.10"]
```

Comment:
DRY fix, but also removed 3.7, and added 3.12 as well as pypy-3.10.
But if we prefer I can address these changes in another dedicated PR